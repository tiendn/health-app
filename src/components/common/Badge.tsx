import React from "react";
import styles from "./style.module.scss";
import { Colors } from "@/theme/colors";
import clsx from "clsx";
import Fonts from "@/theme/fonts";

interface BadgeProps {
	value: number;
	size?: number;
	backgroundColor?: string;
	textColor?: string;
}

const Badge = ({
	value,
	size = 16,
	backgroundColor = Colors.primary[500],
	textColor = Colors.light,
}: BadgeProps) => (
	<div
		className={styles.badge}
		style={{
			width: size,
			height: size,
			borderRadius: size / 2,
			backgroundColor,
			color: textColor,
		}}
	>
		<span className={clsx(styles.badgeValue, Fonts.inter.className)}>{value}</span>
	</div>
);
export default Badge;
