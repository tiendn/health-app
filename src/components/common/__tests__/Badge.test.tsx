import React from "react";
import renderer from "react-test-renderer";
import Badge from "../Badge";

test("Badge renders correctly", () => {
	const tree = renderer.create(<Badge value={3} />).toJSON();
	expect(tree).toMatchSnapshot();
});
