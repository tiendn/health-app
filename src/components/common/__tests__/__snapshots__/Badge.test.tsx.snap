// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly 1`] = `
<div
  className="badge"
  style={
    {
      "backgroundColor": "#EA6C00",
      "borderRadius": 8,
      "color": "#FFFFFF",
      "height": 16,
      "width": 16,
    }
  }
>
  <span
    className="badgeValue className"
  >
    3
  </span>
</div>
`;
